### PR TITLE
New output type: Webex

### PR DESCRIPTION
**Pull request by k0rventen**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area config

/area outputs

> /area tests

**What this PR does / why we need it**:

This PR adds the 'Webex' output, enabling sending messages to a given Webex Room using a Webhook URL. As the format expected by the [Webex API endpoint](https://developer.webex.com/docs/api/v1/messages/create-a-message) is very specific, using the Webhook output isn't possible. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

I've taken inspiration from the Teams output, which is quite similar, which is why I placed the various Webex declarations right below it. If you prefer another order, let me know. 

I've tested it sucessfully using the provided curls's tests commands, and a few falco rules of my own.
Also let me now if I missed some docs/example ? 

Regards !

### CODE REVIEW

**Comment by Issif** [comment]
Amazing proposal, and it seems very complete. I will do the full review asap. Thanks

**Comment by k0rventen** [comment]
> Amazing proposal, and it seems very complete. I will do the full review asap. Thanks

Hey thx ! I just pushed a few fixes, apparently I forgot to check if I changed 'teams' to 'webex' everywhere. Also an improvement on the template loading, which is now done only once instead of on every call of the post function (not a go expert but I think it's better ?)

Cheers !

File: config_example.yaml [hunk 1]
```diff
@@ -65,6 +65,10 @@ teams:
   outputformat: "all" # all (default), text, facts
   minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
 
+webex:
+  #webhookurl: "" # Webex WebhookURL, if not empty, Teams Webex is enabled
+  #minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
```
Comment (Issif) [inline config_example.yaml:70]:
```suggestion
  # webhookurl: "" # Webex WebhookURL, if not empty, Teams Webex is enabled
  # minimumpriority: "" # minimum priority of event for using this output, order is emergency|alert|critical|error|warning|notice|informational|debug or "" (default)
```

File: docs/outputs/webex.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Webex
+
```
Comment (Issif) [inline docs/outputs/webex.md:2]:
```suggestion
```

File: docs/outputs/webex.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  - [Table of content](#table-of-content)
+  - [Configuration](#configuration)
+  - [Example of config.yaml](#example-of-configyaml)
+  - [Screenshots](#screenshots)
+
+## Configuration
+
+| Setting                 | Env var                 | Default value                                                                                       | Description                                                                                                                         |
+| ----------------------- | ----------------------- | --------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------- |
+| `teams.webhookurl`      | `TEAMS_WEBHOOKURL`      |                                                                                                     | Teams WebhookURL, if not empty, Teams output is **enabled** |
```
Comment (Issif) [inline docs/outputs/webex.md:19]:
```suggestion
| `wexeb.webhookurl`      | `WEBEX_WEBHOOKURL`      |                                                                                                     | Teams WebhookURL, if not empty, Webex output is **enabled** |
```

File: outputs/webex.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    * {{ $t }}
+    {{ end }}
+  * Fields:
+	{{ range $key, $value := .OutputFields }}
+	* {{ $key }}: {{ $value }}
+	{{ end }}
+`
+
+// Load the md template
+var tmpl, _ = template.New("markdown").Parse(md)
```
Comment (Issif) [inline outputs/webex.go:32]:
```suggestion
var webexTmpl, _ = template.New("markdown").Parse(md)
```

File: outputs/webex.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+// the format is {"markdown":"..."}
+type webexPayload struct {
+	Markdown string `json:"markdown"`
+}
+
+func newWebexPayload(falcopayload types.FalcoPayload) webexPayload {
+	var tpl bytes.Buffer
+
+	if err := tmpl.Execute(&tpl, falcopayload); err != nil {
```
Comment (Issif) [inline outputs/webex.go:42]:
```suggestion
	if err := webexTmpl.Execute(&tpl, falcopayload); err != nil {
```

**Review by Issif — CHANGES_REQUESTED** [review]

**Comment by k0rventen** [comment]
> could you also squash your commits once you fixed my comments, please

We should be good now !

**Comment by Issif** [comment]
> > could you also squash your commits once you fixed my comments, please
> 
> We should be good now !

Just the squash to do and I'll approve. Thanks

**Comment by Issif** [comment]
> > > > could you also squash your commits once you fixed my comments, please
> > > 
> > > 
> > > We should be good now !
> > 
> > 
> > Just the squash to do and I'll approve. Thanks
> 
> yeah sorry when squashing I rewrote the commit message, leading to this one: [0db1a5f](https://github.com/falcosecurity/falcosidekick/commit/0db1a5f7161495d9add6329a1af75968d1b23a35)

the 6 commits are still there, most of them are typo fixes, rebase on the master then squash (rebase -i) your commits on the HEAD/master please

**Comment by k0rventen** [comment]
it's now all in 19a6aac7688860b778731152d4a8f0ba3cd93acb :smile:

File: docs/outputs/webex.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  - [Table of content](#table-of-content)
+  - [Configuration](#configuration)
+  - [Example of config.yaml](#example-of-configyaml)
+  - [Screenshots](#screenshots)
+
+## Configuration
+
+| Setting                 | Env var                 | Default value                                                                                       | Description                                                                                                                         |
+| ----------------------- | ----------------------- | --------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------- |
+| `wexeb.webhookurl`      | `WEBEX_WEBHOOKURL`      |                                                                                                     | Teams WebhookURL, if not empty, Webex output is **enabled** |
```
Comment (Issif) [inline docs/outputs/webex.md:18]:
```suggestion
| `webex.webhookurl`      | `WEBEX_WEBHOOKURL`      |                                                                                                     | Teams WebhookURL, if not empty, Webex output is **enabled** |
```

File: outputs/webex.go [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+func (c *Client) WebexPost(falcopayload types.FalcoPayload) {
+	c.Stats.Webex.Add(Total, 1)
+
+	err := c.Post(newWebexPayload(falcopayload))
+
+	if err != nil {
+		go c.CountMetric(Outputs, 1, []string{"output:webex", "status:error"})
+		c.Stats.Webhook.Add(Error, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "webex", "status": Error}).Inc()
+		log.Printf("[ERROR] : webex - %v\n", err.Error())
```
Comment (Issif) [inline outputs/webex.go:63]:
```suggestion
		log.Printf("[ERROR] : Webex - %v\n", err.Error())
```

**Review by Issif — CHANGES_REQUESTED** [review]

**Review by Issif — APPROVED** [review]

**Pull request merged**